sources/platform/integrations/programming/api.md

```diff
@@ -164,7 +178,32 @@ If you set up a webhook pointing to the Apify API, the Apify platform will autom
 Therefore, you need to make sure the token has sufficient permissions not only to set up the webhook, but also to perform the actual operation.
 
 :::tip
-
 Let's say you want to create a webhook that pushes an item to a dataset every time an Actor successfully finishes. Then such a scoped token needs to be allowed to both run the Actor (to create the webhook), and write to that dataset.
-
 :::
+
+### Troubleshooting
+
+#### How do I allow a token to run a task?
+
+There is no dedicated run permission for tasks. Instead, configure the token with the following permissions:
+
+- `Run` on the Actor that the task is executing
+- `Read` on the task
+
+See the example below:
+
+![Scoped token configured to run a task](../images/api-token-scoped-run-tasks.png)
+
+Refer to [this section](#permission-dependencies) to understand how permission dependencies work.
+
+#### My run failed and I can see `insufficient permissions` in the logs
+
+You probably run the Actor using a scoped token with `Restricted access` configured.
+
+![Scoped token with Restricted access](../images/api-token-scoped-restricted-access-active.png)
+
+What is happening is that the Actor is trying to access a resource (such a dataset, or a key-value store) or perform an operation that it does not have sufficient permissions for.
+
+If you know what it is, you can add the permission to the scope of your token. If you don't, you can switch the permission mode on the token to `Full access`. This means that the Actor will be able to access all your account data.
+
+Refer to [this section](#actor-execution) to understand how executing Actors with scoped tokens works.
```

Comment:
I recommend avoiding wording as `this` or `here` when using a link due to accessibility reasons.

```suggestion
Refer to [Actor execution](#actor-execution) section to understand how executing Actors with scoped tokens works.
```